pandas/_libs/tslibs/offsets.pyx

```diff
@@ -1221,12 +1221,7 @@ cdef class RelativeDeltaOffset(BaseOffset):
                 # perform calculation in UTC
                 other = other.replace(tzinfo=None)
 
-            if self.n > 0:
-                for i in range(self.n):
-                    other = other + self._offset
-            else:
-                for i in range(-self.n):
-                    other = other - self._offset
+            other = other + (self._offset * self.n)
```

Comment:
sorry, I hadn't noticed this yesterday, but the code change here affects both `self.n > 0` and `self.n <= 0`. however, the test cases we've added only deal with the `self.n > 0` case

seeing as you've already parametrised the test (nice one!), would it be possible to add test cases for `n=0` and `n<0`? just to be safe

Reply:
My bad! I should have added all the scenarios. Thanks for pointing it out. Added them now! Do let me know if there's anything else. @MarcoGorelli